base_layer/wallet/src/storage/sqlite_db.rs

```
@@ -204,33 +229,99 @@ impl WalletSqliteDatabase {
         }
         Ok(())
     }
+
+    fn get_comms_secret_key(&self, conn: &SqliteConnection) -> Result<Option<DbValue>, WalletStorageError> {
+        let cipher = acquire_read_lock!(self.cipher);
+        if let Some(key_str) = WalletSettingSql::get(DbKey::CommsSecretKey.to_string(), &conn)? {
+            let secret_key = match cipher.as_ref() {
+                None => CommsSecretKey::from_hex(key_str.as_str())?,
+                Some(cipher) => {
+                    let decrypted_key_bytes = decrypt_bytes_integral_nonce(&cipher, from_hex(key_str.as_str())?)
+                        .map_err(|_| WalletStorageError::AeadError("Decryption Error".to_string()))?;
```

Still swallowing this error

---

I removed these all as well.